internal/cli/user/profile_test.go

```diff
@@ -1,57 +1,58 @@
-package user_test
+package user
 
 import (
-	"fmt"
-	"io/ioutil"
-	"strings"
 	"testing"
 
-	"github.com/10gen/realm-cli/internal/cli/user"
-	u "github.com/10gen/realm-cli/internal/utils/test"
+	"github.com/10gen/realm-cli/internal/telemetry"
 	"github.com/10gen/realm-cli/internal/utils/test/assert"
+
+	"go.mongodb.org/mongo-driver/bson/primitive"
 )
 
-func TestProfile(t *testing.T) {
-	tmpDir, teardownTmpDir, tmpDirErr := u.NewTempDir("home")
-	assert.Nil(t, tmpDirErr)
-	defer teardownTmpDir()
+func TestProfileResolveFlags(t *testing.T) {
+	t.Run("should provide defaults if flags are empty and set them in the profile", func(t *testing.T) {
+		profile, err := NewProfile(primitive.NewObjectID().Hex())
+		assert.Nil(t, err)
 
-	_, teardownHomeDir := u.SetupHomeDir(tmpDir)
-	defer teardownHomeDir()
+		assert.Equal(t, telemetry.ModeEmpty, profile.Flags.TelemetryMode)
+		assert.Equal(t, "", profile.Flags.RealmBaseURL)
+		assert.Equal(t, "", profile.Flags.AtlasBaseURL)
 
-	profile, profileErr := user.NewDefaultProfile()
-	assert.Nil(t, profileErr)
+		assert.Nil(t, profile.ResolveFlags())
 
-	t.Run("Should initialize as an empty, default profile", func(t *testing.T) {
-		assert.Equal(t, user.DefaultProfile, profile.Name)
-		assert.Equal(t, tmpDir+"/.config/realm-cli", profile.Dir())
-	})
+		assert.Equal(t, telemetry.ModeEmpty, profile.Flags.TelemetryMode)
+		assert.Equal(t, telemetry.ModeEmpty, profile.TelemetryMode())
 
```

Comment:
question out of curiosity: what is this TelemetryMode?

Reply:
the "telemetry" for the CLI is what connects to our Segment service which then sends events as data.  these events have data like `userId`, `command`, etc. so product can better understand how people are using our tools.  we do similar tracking in the UI (for various button clicks) and the backend as well.

the "mode" for telemetry here is basically allowing people to turn that tracking on/off.  similar to those popups you occasionally see asking you to allow an app to report usage data and metrics.